### PR TITLE
V bpastiu/issue10

### PR DESCRIPTION
**Pull request by v-bpastiu**
- adds dev ingress section
- adds cert manager, ingress manifest files

### CODE REVIEW

File: docs/02-development/setup-ingress.md [hunk 1]
```diff
@@ -1 +1,125 @@
...
+
+    !!! info
+        Please note that this domain matches the domain you purchased in the `Prerequisites` section. You will use this domain to create additional sub-domains to use with the microservices app you will deploy in this section.
+
+4. Create an `A` record for your host:
+
+    ```shell
+    LOAD_BALANCER_IP=$(doctl compute load-balancer list --format IP --no-header)
+    doctl compute domain records create <YOUR_DOMAIN_NAME> --record-type "A" --record-name <YOUR_RECORD_NAME> --record-data "$LOAD_BALANCER_IP" --record-ttl "30"
+    ```
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:49]: Please split long lines like this:

```
doctl compute domain records create <YOUR_DOMAIN_NAME> \
        --record-type "A" --record-name <YOUR_RECORD_NAME> \
        --record-data "$LOAD_BALANCER_IP" \
        --record-ttl "30"
```

Also, explain to the user that the above snippet will work only if he/she has **only one load balancer**.

File: docs/02-development/setup-ingress.md [hunk 2]
```diff
@@ -1 +1,125 @@
...
+    helm search repo ingress-nginx
+    ```
+
+2. Install the `Nginx Ingress Controller` using `Helm`:
+
+    ```shell
+    helm install ingress-nginx ingress-nginx/ingress-nginx --version 4.1.3 \
+    --namespace ingress-nginx \
+    --create-namespace  
+    ```
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:30]: Make sure to indent following lines like this:

```
helm install ingress-nginx ingress-nginx/ingress-nginx --version 4.1.3 \
        --namespace ingress-nginx \
        --create-namespace
```

File: docs/02-development/setup-ingress.md [hunk 3]
```diff
@@ -1 +1,125 @@
...
+
+6. Install the `jetstack/cert-manager` chart using `Helm`:
+
+    ```shell
+    helm install cert-manager jetstack/cert-manager --version 1.8.0 \
+    --namespace cert-manager \
+    --create-namespace \
+    --set installCRDs=true
+    ```
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:66]: Make sure to indent following lines like this:

```
    helm install cert-manager jetstack/cert-manager --version 1.8.0 \
        --namespace cert-manager \
        --create-namespace \
        --set installCRDs=true
```

File: docs/02-development/setup-ingress.md [hunk 4]
```diff
@@ -1 +1,125 @@
-TBD.
+## Introduction
+
+In this section, you will learn how to use the Kubernetes-maintained [Nginx](https://kubernetes.github.io/ingress-nginx) Ingress Controller. Then, you're going to discover how to have `TLS` certificates automatically deployed and configured for your hosts (thus enabling `TLS` termination), and `route` traffic to your `backend` applications.
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:3]: Maybe rephrase and simplify a little bit like this:

> In this section, you will learn how to install and configure the Kubernetes-maintained version of the [Nginx](https://kubernetes.github.io/ingress-nginx) Ingress Controller. Then, you're going to discover how to automatically issue `TLS` certificates for your hosts (thus enabling `TLS` termination), and `route` traffic to your `backend` applications.

File: docs/02-development/setup-ingress.md [hunk 5]
```diff
@@ -1 +1,125 @@
...
+
+## Prerequisites
+
+1. A container registry already set up as explained in the [Set up DOCR](setup-docr.md) section.
+2. A Kubernetes cluster (DOKS) up and running as explained in the [Set up DOKS](setup-doks.md) section.
+3. The microservices application deployed in your development cluster as explained in the [Tilt remote](tilt-remote.md) section.
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:9]: Maybe:

> 3. The online boutique sample application deployed to your cluster as explained in the [Tilt remote development](tilt-remote.md#remote-development-with-tilt) section.

File: docs/02-development/setup-ingress.md [hunk 6]
```diff
@@ -1 +1,125 @@
...
+4. A valid domain available configured to point to DigitalOcean name servers. More information in this [article](https://www.digitalocean.com/community/tutorials/how-to-point-to-digitalocean-nameservers-from-common-domain-registrars). Digital Ocean is not a domain registrar so you will need to purchase that from another vendor.
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:10]: Maybe:

> 4. A valid domain available and configured to point to DigitalOcean name servers. More information is available in this [article](https://www.digitalocean.com/community/tutorials/how-to-point-to-digitalocean-nameservers-from-common-domain-registrars). Digital Ocean is not a domain registrar, so you will need to purchase the domain from a well known vendor, such as [GoDaddy](https://www.godaddy.com/).

File: docs/02-development/setup-ingress.md [hunk 7]
```diff
@@ -1 +1,125 @@
...
+
+## Installing ingress nginx
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:12]: Maybe:

> ## Installing the Nginx Ingress Controller

File: docs/02-development/setup-ingress.md [hunk 8]
```diff
@@ -1 +1,125 @@
...
+
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:13]: Add a short description before enumerating the steps - e.g. :

> In this section you will install the community maintained version of the Nginx ingress controller. Please follow below steps to install Nginx using Helm:

File: docs/02-development/setup-ingress.md [hunk 9]
```diff
@@ -1 +1,125 @@
...
+In this section, you will learn how to use the Kubernetes-maintained [Nginx](https://kubernetes.github.io/ingress-nginx) Ingress Controller. Then, you're going to discover how to have `TLS` certificates automatically deployed and configured for your hosts (thus enabling `TLS` termination), and `route` traffic to your `backend` applications.
+
+## Prerequisites
+
+1. A container registry already set up as explained in the [Set up DOCR](setup-docr.md) section.
+2. A Kubernetes cluster (DOKS) up and running as explained in the [Set up DOKS](setup-doks.md) section.
+3. The microservices application deployed in your development cluster as explained in the [Tilt remote](tilt-remote.md) section.
+4. A valid domain available configured to point to DigitalOcean name servers. More information in this [article](https://www.digitalocean.com/community/tutorials/how-to-point-to-digitalocean-nameservers-from-common-domain-registrars). Digital Ocean is not a domain registrar so you will need to purchase that from another vendor.
+
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:11]: We should add Helm, from the installing tools chapter as a requirement as well.

File: docs/02-development/setup-ingress.md [hunk 10]
```diff
@@ -1 +1,125 @@
...
+    ```shell
+    helm install ingress-nginx ingress-nginx/ingress-nginx --version 4.1.3 \
+    --namespace ingress-nginx \
+    --create-namespace  
+    ```
+
+    !!! note
+        To check if the installation was succesfull you can run the `helm ls -n ingress-nginx` and confirm the deployment status.
+
+3. Confiure DNS for Nginx Ingress Controller:
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:35]: Typo - `Confiure -> Configure`.

> 3. Configure DNS for Nginx ingress controller:

File: docs/02-development/setup-ingress.md [hunk 11]
```diff
@@ -1 +1,125 @@
...
+
+    ```shell
+    doctl compute domain create <YOUR_DOMAIN_NAME>
+    ```
+
+    !!! info
+        Please note that this domain matches the domain you purchased in the `Prerequisites` section. You will use this domain to create additional sub-domains to use with the microservices app you will deploy in this section.
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:42]: Why not use a link to the Prerequisites section, instead of highlighting?

File: docs/02-development/setup-ingress.md [hunk 12]
```diff
@@ -1 +1,125 @@
...
+
+    ```shell
+    helm install cert-manager jetstack/cert-manager --version 1.8.0 \
+    --namespace cert-manager \
+    --create-namespace \
+    --set installCRDs=true
+    ```
+
+    !!! note
+        To check if the installation was succesfull you can run the `helm ls -n cert-manager` and confirm the deployment status.
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:69]: Maybe:

> To check if the installation was successful, run the `helm ls -n cert-manager` command, and confirm the deployment status.

File: docs/02-development/setup-ingress.md [hunk 13]
```diff
@@ -1 +1,125 @@
...
+    ```shell
+    DO_API_TOKEN="<YOUR_DO_API_TOKEN_HERE>"
+
+    kubectl create secret generic "digitalocean-dns" \
+        --namespace microservides-demo-dev \
+        --from-literal=access-token="$DO_API_TOKEN"
+    ```
+
+    !!! note
+        The secret must be created in the **same namespace** where the `Issuer` CRD is located in this case the `microservides-demo-dev` namespace.
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:82]: Maybe:

>The secret must be created in the **same namespace** where the `Issuer` CRD is located - in this case the `microservides-demo-dev` namespace.

File: docs/02-development/setup-ingress.md [hunk 14]
```diff
@@ -1 +1,125 @@
...
+
+8. Add the `Cert Manager` issuer using `kubectl` and the provided manifest file:
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:84]: Maybe:

>8. Create an `issuer` resource for cert-manager using `kubectl` (make sure to replace the `<>` placeholders first):

Also, we need to let the user preview the content of the manifest so that he/she can see what is going to be created before running the command. I think we can leverage the power of [collapsible blocks](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#collapsible-blocks) - what do you think ?

File: docs/02-development/setup-ingress.md [hunk 15]
```diff
@@ -1 +1,125 @@
...
+
+    ```shell
+    kubectl apply -f assets/manifests/cert-manager-wildcard-issuer.yaml
+    ```
+
+    !!! info
+        Make sure to replace the `<>` placeholders. Running `kubectl get issuer letsencrypt-nginx-wcard -n microservices-demo-dev` should result in the `True` value being displayed under the `READY` column.
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:91]: The "make sure to replace the .. bla bla" should be stated earlier in the process imho (exemplified in my previous comment).

File: docs/02-development/setup-ingress.md [hunk 16]
```diff
@@ -1 +1,125 @@
...
+
+    !!! note
+        If the `Issuer` object displays a `Not Ready` state you can describe the object to get additional information using: `kubectl describe issuer letsencrypt-nginx-wcard -n microservices-demo-dev`.
+
+9. Add the `Wildcard Certificates` using `kubectl` and the provided manifest file:
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:96]: Don't use capitalised words here.

File: docs/02-development/setup-ingress.md [hunk 17]
```diff
@@ -1 +1,125 @@
...
+10. Add the `Ingress Nginx` host using `kubectl` and the provided manifest file:
+
+    ```shell
+    kubectl apply -f assets/manifests/ingress-host.yaml 
+    ```
+
+    !!! info
+        Make sure to replace the `<>` placeholders.
+
+11. Open a web browser and point to `<YOUR_A_RECORD>.<YOUR_DOMAIN>`. You should see the online boutique welcome page. You should see that the connection is secure and the certificate is a valid one issued by [Let's Encrypt](https://letsencrypt.org).
```
Comment (v-ctiutiu) [inline docs/02-development/setup-ingress.md:123]: Too many "you should see" in one place.

**Review by v-ctiutiu — CHANGES_REQUESTED** [review]

File: docs/02-development/setup-ingress.md (hunk 1 above)
Comment (v-bpastiu) [reply docs/02-development/setup-ingress.md:49]: Fixed in the latest commit.

File: docs/02-development/setup-ingress.md (hunk 2 above)
Comment (v-bpastiu) [reply docs/02-development/setup-ingress.md:30]: Fixed in the latest commit

File: docs/02-development/setup-ingress.md (hunk 3 above)
Comment (v-bpastiu) [reply docs/02-development/setup-ingress.md:66]: Fixed in the latest commit.

File: docs/02-development/setup-ingress.md (hunk 4 above)
Comment (v-bpastiu) [reply docs/02-development/setup-ingress.md:3]: Fixed in the latest commit.

File: docs/02-development/setup-ingress.md (hunk 5 above)
Comment (v-bpastiu) [reply docs/02-development/setup-ingress.md:9]: Fixed in the latest commit.

File: docs/02-development/setup-ingress.md (hunk 8 above)
Comment (v-bpastiu) [reply docs/02-development/setup-ingress.md:13]: Fixed in latest commit.

File: docs/02-development/setup-ingress.md (hunk 9 above)
Comment (v-bpastiu) [reply docs/02-development/setup-ingress.md:11]: Done

File: docs/02-development/setup-ingress.md (hunk 16 above)
Comment (v-bpastiu) [reply docs/02-development/setup-ingress.md:96]: Done

File: docs/02-development/setup-ingress.md (hunk 14 above)
Comment (v-bpastiu) [reply docs/02-development/setup-ingress.md:84]: Looks a lot better. Please see the latest commit.

File: docs/02-development/setup-ingress.md (hunk 15 above)
Comment (v-bpastiu) [reply docs/02-development/setup-ingress.md:91]: Looks a lot better. Please see the latest commit.
